### PR TITLE
feat: query smart wallet auto-provision fee

### PR DESCRIPTION
**Pull request by samsiegart**
refs https://github.com/Agoric/ui-kit/issues/78

This queries the smart wallet auto-provision fee in the wallet connection component, and exposes it in the react `useAgoric` hook as well.

Also fixed a bug where the purse notifier was updating every time the bank was polled, even if the balances didn't change. (cc @0xpatrickdev)

## Testing

Tested on this branch of dapp-offer-up: https://github.com/Agoric/dapp-offer-up/compare/test-network-dropdown

Steps:
- Build ui-kit with `yarn && yarn workspaces run prepack`
- Set up link with `cd packages/react-components && yarn link`
- Clone `dapp-offer-up` and checkout this branch: https://github.com/Agoric/dapp-offer-up/compare/test-network-dropdown
- Start the local chain with docker (the offer-up contract doesn't necessarily have to be deployed to test this functionality, just a local chain)
- Link the local version of this packages in dapp-offer-up: `cd ui && yarn link @agoric/react-components`
- Run the local UI: `cd ui && yarn && yarn dev`
- Try modifying the dapp to read the provision fee: `const { smartWalletProvisionFee } = useAgoric()`
- Observe after connecting to wallet that the provision fee is accessible

For an e2e example with the provision dialog, see https://github.com/Agoric/ui-kit/pull/99

### CODE REVIEW

**Comment by LuqiPan** [comment]
@samsiegart could you help me and other reviewers understand what tests have you done on this PR?

**Comment by samsiegart** [comment]
> @samsiegart could you help me and other reviewers understand what tests have you done on this PR?

Sure yea, just updated the PR description. Lmk if you'd like more clarification or want to walk through it together. We're pretty close to being able to set up an example dapp in this repo with e2e testing, which would make this type of thing easier to test.

File: packages/web-components/package.json [hunk 1]
```diff
@@ -17,6 +17,7 @@
     "@agoric/assert": "^0.6.0",
     "@agoric/cache": "^0.3.2",
     "@agoric/casting": "^0.4.3-u13.0",
+    "@agoric/cosmic-proto": "0.3.0",
```
Comment (turadg) [inline packages/web-components/package.json:20]: does it work with 0.4.0?

File: packages/web-components/src/wallet-connection/querySwingsetParams.js [hunk 2]
```diff
@@ -0,0 +1,18 @@
+import { QueryClient, createProtobufRpcClient } from '@cosmjs/stargate';
+import { QueryClientImpl } from '@agoric/cosmic-proto/swingset/query.js';
+
+/** @typedef {import('@agoric/cosmic-proto/dist/agoric/swingset/query').QueryParamsResponse} QueryParamsResponse */
```
Comment (turadg) [inline packages/web-components/src/wallet-connection/querySwingsetParams.js:4]: try TS 5.5 beta for real imports. I'll send a PR for that

File: packages/web-components/package.json (hunk 1 above)
Comment (samsiegart) [reply packages/web-components/package.json:20]: Following up, looks like 0.4.0 isn't published, tried `0.4.1-dev-e2e36cc.0` but doesn't seem to have the correct interface, and best not to risk https://github.com/Agoric/dapp-inter/issues/252. Will keep at 0.3.0 for now.

File: packages/web-components/src/wallet-connection/watchWallet.js [hunk 3]
```diff
@@ -79,24 +82,86 @@ export const watchWallet = (
...
+          }
+
+          const fee =
+            (BigInt(beansPerSmartWallet) / BigInt(feeUnit)) *
+            BigInt(feeUnitPrice);
+
+          res(fee);
+        } catch (e) {
+          console.error('Error querying smart wallet provision fee', address);
+          if (attempts >= MAX_ATTEMPTS_TO_WATCH_BANK) {
```
Comment (turadg) [inline packages/web-components/src/wallet-connection/watchWallet.js:127]: retry logic seems ripe for generalization. e.g. https://caolan.github.io/async/v2/retry.js.html

or this light package https://github.com/vercel/async-retry

**Review by turadg — APPROVED** [review]

**Pull request merged**